core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -528,6 +548,16 @@ class StandaloneDynamicAllocationSuite
     }
   }
 
+  /** Kill the executors on a given host. */
+  private def killExecutorsOnHost(sc: SparkContext, host: String): Boolean = {
+    syncExecutors(sc)
+    sc.schedulerBackend match {
+      case b: CoarseGrainedSchedulerBackend =>
+        b.killExecutorsOnHost(host, false)
```

Comment:
Yes, the test was the sole reason for exposing replace, so it can be removed. I was having some issues with .keySet earlier, and, we do need to `syncExecutors(sc)` after killing to pass the test. I'm keeping the `eventually` block since killing happens asynchronously.